File: sohy/190724_rotate_string.java
class Solution {
    public boolean rotateString(String A, String B) {
        int lenA = A.length();
        int lenB = B.length();
        if (lenA != lenB) {
            return false; 
        } else if (lenA == 0 && lenB == 0) {
            return true; 
        }
        
        
        int count = 0; 
        while (count < lenA) {
            char current = A.charAt(0);
            A = A.substring(1) + current; 
            System.out.println(A);
            if (A.equals(B)) {
                return true; 
            }
            count ++; 
        }
        return false; 
        
    }
}

/*
Runtime: 8 ms, faster than 5.12% of Java online submissions for Rotate String.
Memory Usage: 36.1 MB, less than 34.32% of Java online submissions for Rotate String.

-> need to optimize 

*/ 

File: 2kindsofcs/190724-rotate-string.py
class Solution:
    def rotateString(self, A: str, B: str) -> bool:
        if not A and not B:
            return True
        if A == B:
            return True
        AList = [ letter for letter in A ]
        BList = [ letter for letter in B ]
        ALength = len(A)
        BLength = len(B)
        if ALength != BLength:
            return False 
        for i in range(ALength):
            AList.append(AList.pop(0))
            if AList == BList:
                return True
        return False 
        
# Runtime: 32 ms, faster than 91.27% of Python3 online submissions for Rotate String.
# Memory Usage: 13.6 MB, less than 5.42% of Python3 online submissions for Rotate String.
